python-avd/pyavd/_cv/workflows/models.py

```diff
@@ -177,6 +233,66 @@ class CVTimeOuts:
     change_control_creation_timeout: float = 300.0
 
 
+@dataclass
+class WorkflowDevice:
+    """Internal mutable representation of a device for deployment workflow with inputs from AVD."""
+
+    avd_device: AvdDevice
+    result: DeviceResult
+    in_cv_inventory: bool = False
+    serial_number: str | None = None
+    """Device serial number retrieved from CV"""
+    system_mac_address: str | None = None
+    """Device system mac address retrieved from CV"""
+
+    def get_cv_device(self) -> CVDevice:
+        # CVDevice object populated with hostname from AVD input and serial number,
+        # system mac address derived from CV Inventory
+        return CVDevice(
+            hostname=self.avd_device.hostname, serial_number=self.serial_number, system_mac_address=self.system_mac_address, _exists_on_cv=self.in_cv_inventory
+        )
+
+    def get_cv_eos_config(self, config_object: AvdEosConfig) -> CVEosConfig:
+        return CVEosConfig(file=config_object.file, device=self.get_cv_device(), configlet_name=config_object.configlet_name)
+
+    def get_one_cv_tag(self, tag_object: AvdDeviceTag | AvdInterfaceTag) -> CVDeviceTag | CVInterfaceTag:
+        # caller to ensure tag_object is valid
+        if isinstance(tag_object, AvdDeviceTag):
+            return self.get_one_cv_device_tag(tag_object)
+        return self.get_one_cv_interface_tag(tag_object)
+
+    def get_one_cv_device_tag(self, avd_device_tag: AvdDeviceTag) -> CVDeviceTag:
+        # caller to ensure avd_device_tag is valid
+        return CVDeviceTag(label=avd_device_tag.label, value=avd_device_tag.value, device=self.get_cv_device())
+
+    def get_one_cv_interface_tag(self, avd_interface_tag: AvdInterfaceTag) -> CVInterfaceTag:
+        # caller to ensure avd_interface_tag is valid
+        return CVInterfaceTag(label=avd_interface_tag.label, value=avd_interface_tag.value, device=self.get_cv_device(), interface=avd_interface_tag.interface)
+
+    def get_cv_pathfinder_metadata(self, pathfinder_metadata: AvdPathfinderMetadata) -> CVPathfinderMetadata:
+        return CVPathfinderMetadata(metadata=pathfinder_metadata.metadata, device=self.get_cv_device())
+
+
+T_DeployElement = TypeVar("T_DeployElement", AvdEosConfig, AvdDeviceTag, AvdInterfaceTag, AvdPathfinderMetadata)
+
+
+@dataclass
+class DeploymentResult(Generic[T_DeployElement]):
+    deployed: list[T_DeployElement] = field(default_factory=list)
+    skipped: list[T_DeployElement] = field(default_factory=list)
+    removed: list[T_DeployElement] = field(default_factory=list)
+
+
+@dataclass
+class DeviceResult:
+    """Device specific results from deployment to CV."""
+
+    device_tags: DeploymentResult[AvdDeviceTag]
+    interface_tags: DeploymentResult[AvdInterfaceTag]
+    config: DeploymentResult[AvdEosConfig]
+    pathfinder_metadata: DeploymentResult[AvdPathfinderMetadata]
```

Comment:
Related to my comment here: https://github.com/aristanetworks/avd/pull/5679/files#r2383804793

Let's replace this by an Enum:

```suggestion
class DeploymentStatus(Enum):
    PENDING = 1
    DEPLOYED = 2
    SKIPPED = 3
    REMOVED = 4
```
Then we can create the appropriate `WorkflowXXX` stateful models for all deployment objects. See the other comments in this module.